### ISSUE TITLE
Delete all cookies on exit?

### ISSUE THREAD

**Issue by cranes-bill** [QupZilla/qupzilla#1901]
My lovely :blue_heart: **v1.8 QupZilla** has an option to delete all cookies on exit. Is there any chance this option to be ported in **QupZilla 2** as well:question:

**Comment by Martii** [comment]
Can't speak for the HEAD but 1.8.9 *(QtWebKit)* seems to have this under Edit &rarr; Privacy &rarr; Cookies Manager &rarr; Settings then Delete cookies on close ticked. Didn't work on close for me though.

**Comment by cranes-bill** [comment]
> Can't speak for the HEAD but 1.8.9 (QtWebKit) seems to have this ...

That's what i meant with „_My lovely  v1.8 QupZilla_“ and it works flawlessly to me.

**Comment by Martii** [comment]
Yup... `Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) QupZilla/1.9.99 Chrome/45.0.2454.101 Safari/537.36` current HEAD is missing this.

**Comment by nowrep** [comment]
Right now there is also no way to disable/filter cookies. The API was removed and will be added (I hope) back for Qt 5.7.
So unless it is changed, I won't bring this feature back.

**Comment by cranes-bill** [comment]
OK, thanks! I'm not sure should i close the issue because you added a label, so reopen it if needed.

**Comment by nowrep** [comment]
Leave it open, or I'll forget about it.

**Comment by Martii** [comment]
@nowrep 
Is there a way to call Tools &rarr; Clear Recent History on exit with a checkbox somewhere? e.g. the routines behind the dialog?

``` sh-session
Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) QupZilla/2.0.0 Chrome/45.0.2454.101 Safari/537.36
```

**Comment by FrictionCow** [comment]
Yes, I'd really like this feature. Once QtWebEngine hopefully adds the API again, would it be possible to Whitelist some cookies to be saved? I'm thinking of a functionality like the addon, Self-Destructing Cookies for Firefox.

**Comment by Chromace** [comment]
There is a workaround currently (at least on Linux). Just navigate to '.config/qupzilla/profiles/[current profile]/', and replace the 'Cookies' database file with a folder named 'Cookies'.

That lets cookies stay for the session, but disappear on closing Qupzilla (I think it has something to do with trying to write to the directory file, but failing since it's a directory). I assume making the 'Cookies' file read only in Windows/OS X would achieve a similar thing.

**Comment by nowrep** [comment]
Fixed 7738deb721631ebb25bbfab25f284caef66f39b1

**Comment by Arnaudv6** [comment]
Thanks !
